src/es/barmanga/src/eu/kanade/tachiyomi/extension/es/barmanga/BarManga.kt

```diff
@@ -15,32 +22,102 @@ class BarManga : Madara(
 ) {
     override val useLoadMoreRequest = LoadMoreStrategy.Never
 
-    override val mangaDetailsSelectorDescription = "div.flamesummary > div.manga-excerpt"
+    override fun mangaDetailsParse(document: Document): SManga {
+        return super.mangaDetailsParse(document).apply {
+            extractRealTitle(document)?.let { title = it }
+            extractRealCover(document)?.let { thumbnail_url = it }
+        }
+    }
+    private fun extractRealTitle(document: Document): String? {
+        return document.selectFirst("span[property=name]")?.text()
+            ?: document.selectFirst("ol.breadcrumb li:last-child a")?.text()
+    }
+
+    private fun extractRealCover(document: Document): String? {
+        document.select("script:containsData(img.src)")
+            .firstNotNullOfOrNull { IMG_SRC_REGEX.find(it.html())?.groupValues?.get(1) }
+            ?.let { return it }
+
+        val style = document.selectFirst("div.profile-manga")?.attr("style") ?: return null
+        return BG_URL_REGEX.find(style)?.groupValues?.get(1)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        document.select("div[style*='display: none'], div[style*='display:none']").remove()
+        return document.select(chapterListSelector()).map { chapterFromElement(it) }
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        return SChapter.create().apply {
+            setUrlWithoutDomain(extractChapterUrl(element))
+            name = extractChapterName(element)
+            date_upload = extractChapterDate(element)
+        }
+    }
 
-    override val pageListParseSelector = "div.page-break"
+    private fun extractChapterUrl(element: Element): String {
+        element.selectFirst("span.chapter-text-content")
+            ?.attr("data-original-href")
+            ?.takeIf { it.isNotEmpty() }
+            ?.let { return it.decodeBase64() }
 
-    private val imageSegmentsRegex = """var\s+imageSegments\s*=\s*\[\s*(['"][A-Za-z0-9+/=]+['"](?:\s*,\s*['"][A-Za-z0-9+/=]+['"])*)\s*];""".toRegex()
-    private val base64ItemRegex = """['"]([A-Za-z0-9+/=]+)['"]""".toRegex()
+        element.attributes().asList()
+            .find { it.key.startsWith("data-") && it.key.endsWith("-url") }
+            ?.value
+            ?.takeIf { it.isNotEmpty() }
+            ?.let { return it.decodeBase64() }
+
+        return element.selectFirst("a")?.attr("abs:href") ?: ""
+    }
+
+    private fun extractChapterName(element: Element): String {
+        return element.selectFirst("span.chapter-text-content")?.text()?.trim()
+            ?: element.selectFirst("a")?.text()?.trim()
```

Comment:
`.text` is already trimmed.

```suggestion
        return element.selectFirst("span.chapter-text-content")?.text()
            ?: element.selectFirst("a")?.text()
```